pyuvsim/simsetup.py

```diff
@@ -1218,12 +1218,13 @@ def parse_telescope_params(tele_params, config_path='', freq_range=None, force_b
     if 'array_layout' not in tele_params:
         raise KeyError('array_layout must be provided.')
     array_layout = tele_params.pop('array_layout')
+    assert isinstance(array_layout, (str, dict)), "array_layout must be a string or a dict"
```

Comment:
I feel like with how much other manual error raising, it might be more meaningful to raise something specific and not just use an assert. But I can be convinced.

Reply:
sure, I can add that. I actually added this when I was trying to figure out a test failure and I fixed the thing that had caused this upstream. But I think you could hit this with an error in the param yaml file, so it is a bit user facing.

Reply:
An error in the param yaml reader would be the best place I agree